src/main/kotlin/com/hibob/academy/employeeFeedback/service/ResponseService.kt

```diff
@@ -0,0 +1,41 @@
+package com.hibob.academy.employeeFeedback.service
+
+import com.hibob.academy.employeeFeedback.dao.*
+import jakarta.ws.rs.BadRequestException
+import jakarta.ws.rs.NotFoundException
+import org.springframework.stereotype.Service
+
+@Service
+class ResponseService(private val responseDao: ResponseDao, private val feedbackDao: FeedbackDao) {
+    fun responseLengthValidation(response: String) {
+        val minCharacters = 30
+        val maxCharacters = 500
+
+        if (response.length < minCharacters || response.length > maxCharacters) throw BadRequestException("Invalid response length")
+    }
+
+    fun isWriterIdentified(id: Int?) =
+        id ?: throw BadRequestException("Unable to respond to anonymous feedback")
+
+    fun doesFeedbackExist(feedbackInfo: FeedbackInfo?) =
+        feedbackInfo ?: throw NotFoundException("This feedback doesn't exist")
+
+    fun submitResponse(feedbackId: Int, reviewerId: Int, response: String, companyId: Int): Boolean {
+        val feedbackToRespond = feedbackDao.getFeedbackById(feedbackId, companyId)
```

Comment:
the first part of the function is basically validations? it's better to wrap in a function.
So when you read the submit function it's clear:

- now I'm validating
- no I'm doing an action